plugins/bgnotify/bgnotify.plugin.zsh

```diff
@@ -85,23 +86,15 @@ function bgnotify {
       esac
     fi
 
-    if [[ -z "$term_id" ]]; then
-      terminal-notifier -message "$message" -title "$title" &>/dev/null
-    else
-      terminal-notifier -message "$message" -title "$title" -activate "$term_id" -sender "$term_id" &>/dev/null
-    fi
+    terminal-notifier -message "$message" -title "$title" ${=icon:+-appIcon "$icon"} ${=term_id:+-activate "$term_id" -sender "$term_id"} &>/dev/null
   elif (( ${+commands[growlnotify]} )); then # macOS growl
     growlnotify -m "$title" "$message"
   elif (( ${+commands[notify-send]} )); then
-    if [[ -n $ALACRITTY_WINDOW_ID ]]; then
-        notify-send -i Alacritty "$title" "$message"
```

Comment:
This was added recently in #12045. This PR will no longer show the Alacritty icon, but the user can easily re-add this functionality by adding this in their ~/.zshrc file:

```
function bgnotify_formatted {
  ## $1=exit_status, $2=command, $3=elapsed_time
  [ $1 -eq 0 ] && title="Great success" || title="Command failed"
  bgnotify "$title - took ${3}s" "$2" "Alacritty"
}
plugins=(git bgnotify sudo)
```

There are several reasons why I'm skeptical to hard-code Alacritty here:
* There are [dozens of different terminal emulators](https://en.wikipedia.org/wiki/List_of_terminal_emulators). How many are a reasonable number to hard-code here?
* Can we ensure icons are somewhat reliably installed in accessible places across different distributions and OSes?

Reply:
I agree, let's go for it!

Reply:
Regarding linux: `notify-send` can take a file path or an app name as icon, and we can expect those icons to be in `/usr/share/icons` or `/usr/share/pixmaps`. The issue I see is that there's no reliable way (that I know of) of getting the appropriate icon name for each terminal, and we'd have to maintain a mapping of env vars -> icon names.

Another approach that could work, is to add an icon for bgnotify in this repo, and have that be the default if `icon` is not explicitly set by the user.